Fix rspec deprecation warnings

kiote opened the pull request:
## Why

we have several deprecation warnings from rspec and rails 5 here

kiote committed:
Fix rspec deprecation warnings

  most fixes is about new rspec syntax

kiote committed:
Fix rails 5.1 upgrade warnings

  we had some code, which won't be working under rails 5.1

kiote committed:
Remove git insecure warning

kiote commented on app/models/sms_broker/outgoing_message.rb:
```diff
@@ -17,7 +17,7 @@ class OutgoingMessage < ActiveRecord::Base
     @@after_create_hooks = Array.new
 
-    # :reek:Manual-Dispatch
+    # :reek:ManualDispatch
```
this change is not really connected to the issue, but we had a warning saying "there is no such cop", cause it was wrong-named 🍳

kiote commented on spec/controllers/sms_broker/incoming_messages_controller_spec.rb:
```diff
@@ -56,9 +56,9 @@ module SmsBroker
 
       it 'ensures correct encoding on incoming text' do
         pending('this test is broken')
```
This test is still pending. Still not sure how to fix that (have `Invalid query parameters: Non UTF-8 value:`) error

rogerkk replied:
Not sure either, since it doesn't allow passing ISO-8851-1 at all anymore. Just delete it for now, perhaps it's not even a real problem. :/

kiote commented on spec/jobs/sms_broker/send_sms_job_spec.rb:
```diff
@@ -6,28 +6,33 @@ module SmsBroker
     describe "#perform" do
       it "raises exception if sending of SMS fails" do
         job = SendSmsJob.new(nil)
-        SmsSender.any_instance.stub(:send) { raise }
```
rspec turns to be a bit critical to raising "just an error", so we need to be more specific here

kiote reviewed (commented):
After this changes we'll survive a little bit longer with this gem ;)

rogerkk reviewed (approved):
Perhaps the pending spec could be deleted, but apart from that I guess this is fine! :)

kiote committed:
Remove wrong-encoding testing

  before we had a test for hypothetical situation, when user sends a message in
  a wrong encoding. But this test started to fail with "Invalid query
  parameters: Non UTF-8 value" and we had no cure for this, but test removing
